x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutDataFrameAnalyticsAction.java

```diff
@@ -160,8 +163,11 @@ private void putValidatedConfig(
         if (securityContext != null) {
             useSecondaryAuthIfAvailable(securityContext, () -> {
                 final String username = securityContext.getUser().principal();
+                final String[] sourceIndices = Arrays.stream(preparedForPutConfig.getSource().getIndex())
```

Comment:
```suggestion
                // DFA doesn't support CCS, but if it did it would need this filter, so it's safest to have the filter
                // in place even though it's a no-op.
                // TODO: Remove this filter once https://github.com/elastic/elasticsearch/issues/67798 is fixed.
                final String[] sourceIndices = Arrays.stream(preparedForPutConfig.getSource().getIndex())
```

Reply:
Done.